pkg/member/member_control.go

```diff
@@ -103,15 +105,19 @@ func (m *memberControl) AddMemberAsLearner(ctx context.Context) error {
 
 	memAddCtx, cancel := context.WithTimeout(ctx, EtcdTimeout)
 	defer cancel()
+	start := time.Now()
 	response, err := cli.MemberAddAsLearner(memAddCtx, []string{memberURL})
 	if err != nil {
 		if errors.Is(err, rpctypes.Error(rpctypes.ErrGRPCPeerURLExist)) || errors.Is(err, rpctypes.Error(rpctypes.ErrGRPCMemberExist)) {
 			m.logger.Infof("Member %s already part of etcd cluster", memberURL)
 			return nil
 		}
+		metrics.AddLearnerDurationSeconds.With(prometheus.Labels{metrics.LabelSucceeded: metrics.ValueSucceededFalse}).Observe(time.Since(start).Seconds())
 		return fmt.Errorf("error while adding member as a learner: %v", err)
 	}
 
+	metrics.HasLearner.With(prometheus.Labels{}).Set(1)
```

Comment:
If the member is adding itself as a learner, shouldn't you also set `IsLearner` metric to true here?